lib/src/network_manager_client.dart

```diff
@@ -2496,6 +2496,39 @@ class NetworkManagerClient {
         object ?? _NetworkManagerObject(_bus, DBusObjectPath('/'), {}));
   }
 
+  /// Activates a connection for [device].
+  ///
+  /// A specific [connection] may be specified, or else it is detected automatically.
+  ///
+  /// Note that when activating a wireless connection, the [accessPoint] must be specified.
+  Future<NetworkManagerActiveConnection?> activateConnection(
+      {required NetworkManagerDevice device,
+      NetworkManagerSettingsConnection? connection,
+      NetworkManagerAccessPoint? accessPoint}) async {
+    assert(device.wireless == null || accessPoint != null);
+    if (_manager == null) {
```

Comment:
I would drop this check, as we have the `!` below that throws an exception. I this `_manager` should be non-null and throw an exception if it is missing (doesn't need to be fixed in this PR).